src/rust/jsg-macros/lib.rs

```diff
@@ -187,17 +189,12 @@ pub fn jsg_method(_attr: TokenStream, item: TokenStream) -> TokenStream {
         #[automatically_derived]
         extern "C" fn #callback_name(args: *mut jsg::v8::ffi::FunctionCallbackInfo) {
             let mut lock = unsafe { jsg::Lock::from_args(args) };
-            let args = unsafe { jsg::v8::FunctionCallbackInfo::from_ffi(args) };
+            let mut args = unsafe { jsg::v8::FunctionCallbackInfo::from_ffi(args) };
             #(#unwrap_statements)*
             let this = args.this();
             let self_ = jsg::unwrap_resource::<Self>(&mut lock, this);
-            match self_.#fn_name(#(#arg_refs),*) {
-                Ok(result) => args.set_return_value(result.wrap(&mut lock)),
-                Err(err) => {
-                    println!("Error: {err}");
-                    todo!("{err}");
-                }
-            }
+            let result = self_.#fn_name(#(#arg_refs),*);
+            unsafe { jsg::handle_result(&mut lock, &mut args, result) };
```

Comment:
Could we put these kinds of unsafe / system calls into a special namespace that is clear that it isn't meant to be used directly in user code?

Reply:
Isn't that the whole point of using `unsafe` keyword in Rust?